### ISSUE TITLE
Version 2.0.0 crashes immediately after startup

### ISSUE THREAD

**Issue by freebrowser1** [bk138/droidVNC-NG#157]
If you'd like to put out an incentive for fixing this bug, you can do so at https://issuehunt.io/r/bk138/droidVNC-NG

**Describe the bug**
I had version 1.4.0 and installed 2.0.0 (downloaded from Github) over it. Afterwards I installed 1.4.0 over 2.0.0 and it worked again normally.

**To Reproduce**

1. Install 2.0.0 over 1.4.0 (the latter worked).
2. Started droidVNC.
3. It crashes and closes.

**Expected Behavior**
It should work normally.

**Your environment (please complete the following information):**
 - droidVNC-NG version: 2.0.0
 - Android version: 10
 - Client-side OS and version: macOS Ventura
 - VNC client and version:RealVNC viewer

**Comment by bk138** [comment]
Thanks for your report! Can you please either:

* install 2.0.0 from google play and repro the crash
* or let 2.0.0 crash and provide an `adb logcat` output

?

**Comment by freebrowser1** [comment]
`adb logcat` shows this about crashing droidvnc:

```
--------- beginning of crash
08-07 21:11:05.182 31549 31549 E AndroidRuntime: FATAL EXCEPTION: main
08-07 21:11:05.182 31549 31549 E AndroidRuntime: Process: net.christianbeier.droidvnc_ng, PID: 31549
08-07 21:11:05.182 31549 31549 E AndroidRuntime: java.lang.RuntimeException: Unable to start service net.christianbeier.droidvnc_ng.MainService@8325e16 with Intent { act=action_handle_media_projection_result cmp=net.christianbeier.droidvnc_ng/.MainService (has extras) }: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4222)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.app.ActivityThread.access$2100(ActivityThread.java:231)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1984)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7682)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: Caused by: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.util.SparseArray.keyAt(SparseArray.java:301)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.net.nsd.NsdManager.getListenerKey(NsdManager.java:490)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.net.nsd.NsdManager.unregisterService(NsdManager.java:574)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.registerNSD(MainService.java:732)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.onStartCommand(MainService.java:275)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4204)
08-07 21:11:05.182 31549 31549 E AndroidRuntime: 	... 8 more
08-07 21:11:13.289 31673 31673 E AndroidRuntime: FATAL EXCEPTION: main
08-07 21:11:13.289 31673 31673 E AndroidRuntime: Process: net.christianbeier.droidvnc_ng, PID: 31673
08-07 21:11:13.289 31673 31673 E AndroidRuntime: java.lang.RuntimeException: Unable to start service net.christianbeier.droidvnc_ng.MainService@c18a6bb with Intent { act=action_handle_media_projection_result cmp=net.christianbeier.droidvnc_ng/.MainService (has extras) }: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4222)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.app.ActivityThread.access$2100(ActivityThread.java:231)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1984)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7682)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: Caused by: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.util.SparseArray.keyAt(SparseArray.java:301)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.net.nsd.NsdManager.getListenerKey(NsdManager.java:490)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.net.nsd.NsdManager.unregisterService(NsdManager.java:574)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.registerNSD(MainService.java:732)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.onStartCommand(MainService.java:275)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4204)
08-07 21:11:13.289 31673 31673 E AndroidRuntime: 	... 8 more
08-07 21:11:27.432 31806 31806 E AndroidRuntime: FATAL EXCEPTION: main
08-07 21:11:27.432 31806 31806 E AndroidRuntime: Process: net.christianbeier.droidvnc_ng, PID: 31806
08-07 21:11:27.432 31806 31806 E AndroidRuntime: java.lang.RuntimeException: Unable to start service net.christianbeier.droidvnc_ng.MainService@8ee27d8 with Intent { act=action_handle_media_projection_result cmp=net.christianbeier.droidvnc_ng/.MainService (has extras) }: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4222)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.app.ActivityThread.access$2100(ActivityThread.java:231)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1984)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7682)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: Caused by: java.lang.ArrayIndexOutOfBoundsException: length=11; index=-1
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.util.SparseArray.keyAt(SparseArray.java:301)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.net.nsd.NsdManager.getListenerKey(NsdManager.java:490)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.net.nsd.NsdManager.unregisterService(NsdManager.java:574)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.registerNSD(MainService.java:732)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at net.christianbeier.droidvnc_ng.MainService.onStartCommand(MainService.java:275)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4204)
08-07 21:11:27.432 31806 31806 E AndroidRuntime: 	... 8 more
--------- beginning of system

08-08 15:38:00.074  1129  1129 D ConditionProviders.SCP: onReceive android.app.action.NEXT_ALARM_CLOCK_CHANGED
08-08 15:38:00.074  1129  1129 D ConditionProviders.SCP: evaluateSubscriptionLocked cal=ScheduleCalendar[mDays={1, 2, 3, 4, 5, 6, 7}, mSchedule=ScheduleInfo{days=[1, 2, 3, 4, 5, 6, 7], startHour=22, startMinute=0, endHour=7, endMinute=0, exitAtAlarm=true, nextAlarm=Tue Aug 08 15:38:00 GMT+02:00 2023 (1691501880000)}], now=Tue Aug 08 15:38:00 GMT+02:00 2023 (1691501880074), nextUserAlarmTime=Tue Aug 08 15:43:46 GMT+02:00 2023 (1691502226171)
```

**Comment by bk138** [comment]
Alright, this is already fixed, by b20cb1731ebdf38aa9418e42c6d8eea441f70532. A new release will be out this week!
